client/lib/wp/sync-handler/cache.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+import debugFactory from 'debug';
+import ms from 'ms';
+
+/**
+ * Internal dependencies
+ */
+import warn from 'lib/warn';
+import { getLocalForage } from 'lib/localforage';
+
+/**
+ * Module variables
+ */
+const localforage = getLocalForage();
+const debug = debugFactory( 'calypso:sync-handler:cache' );
+
+const RECORDS_LIST_KEY = 'records-list';
+const SYNC_RECORD_REGEX = /^sync-record-\w+$/;
+const LIFETIME = '2 days';
+
+/**
+ * Check it the given key is a `sync-record` key
+ *
+ * @param {String} key - record key
+ * @return {Boolean} `true` if it's a sync-record-<key>
+ */
+const isSyncRecordKey = key => {
+	const isSyncRecord = SYNC_RECORD_REGEX.test( key );
+
+	if ( isSyncRecord ) {
+		debug( '%o is a sync-record', key );
+	}
+	return isSyncRecord;
+}
+
+export class Cache {
```

Comment:
Since this class isn't encapsulating any state, perhaps change this to a simple object export.